splunk-forwarder/main.tf

```diff
@@ -0,0 +1,113 @@
+resource "aws_iam_role" "lambda_role" {
+  name = "lambda_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": "sdsd"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_cloudwatch" {
+  name        = "cw_for_lambda"
+  description = "Lets the Lambda function write to cloudwatch"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+    ],
+      "Resource": [
+        "arn:aws:logs:*:*:*"
+    ]
+  }
+ ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_s3_puts" {
+  name        = "s3_for_lambda"
+  description = "Lets the Lambda function write to a any bucket"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:*"
+      ],
+      "Effect": "Allow",
+      "Resource": ["arn:aws:s3:::telia-common-logs-prod-application-logs/*"]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_cw_attach" {
+  role       = "${aws_iam_role.lambda_role.name}"
+  policy_arn = "${aws_iam_policy.allow_cloudwatch.arn}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_s3_attach" {
+  role       = "${aws_iam_role.lambda_role.name}"
+  policy_arn = "${aws_iam_policy.allow_s3_puts.arn}"
+}
+
+// Values like bucket names and keys are hard coded here because they depend on already- existing
```

Comment:
I think this all has to be parametrized but it can have default values set to what you already have.